### DISCUSSION TITLE
k3s change ip address

### DISCUSSION THREAD

**Discussion by Heyz6** [k3s-io/k3s#7326] [General]
When I run k3s single node, or a cluster, and the host Ip sends a change, how do I make the change to keep k3s in use?

**Comment by Cnordbo** [comment]
Do you mean how to get it up and running again after the IP has changed? 
I just struggled with the same thing myself while setting up a cluster at my home network. And then only afterwards thinking of setting up a static IP on the master and nodes. 

In my case, every IP changed on both server and agent. 
So im not sure if this is needed if ever just an agent changes its IP.

What worked for me was doing the following:

1. Login to the k3s `server`
2. run `sudo kubectl delete node <name-of-node>` (No need for sudo if your kubeconfig already has `644` as its write permissions)
3. If the server IP changed - Login to the node
4. Change k3S_URL in `/etc/systemd/system/k3s-agent.service.env` file to the correct value

Restart agent. Im not sure if a service restart will do (`systemctl restart k3s-agent`) - I just rebooted the entire agent.

**Comment by Hacker-OH** [comment]
Quick question, are you able to run k3s with a dynamically changing IP address? Say for example you want a worker node to be able to connect to different networks but still talked to the master node.

**Comment by harryzcy** [comment]
I managed to just change `k3S_URL` in `/etc/systemd/system/k3s-agent.service.env` file, and then run `systemctl restart k3s-agent`, and I have the nodes connected again.

I didn't even remove the node, and didn't reboot the system.

**Comment by caroline-suse-rancher** [comment]
Converting this to a discussion

**Comment by sylwester-** [comment]
I’m planning a significant network restructure and would appreciate input on best practices for this transition with k3s.

Background and plans:

- Currently running my k3s cluster on a single flat network.
- I will soon migrate to Ubiquiti/UniFi hardware.
- The goal is to create several VLANs, including a dedicated VLAN for all IoT devices.
- I might have k3s on its own VLAN
- As a result, the k3s cluster (both nodes and master) will receive new IP addresses - and I want this to happen intentionally as part of the migration.

Questions & points for advice:

- What’s the recommended approach for cleanly re-IPing the k3s cluster (master and nodes) during a network migration?
- Is it better to fully remove and re-add the nodes after updating the networks and VLAN assignments, or are there other techniques that mitigate downtime and messy node records?

**Reply by Lywel** [reply]
I'm in the same situation. What did you end up doing ?

**Reply by sylwester-** [reply]
I ended up backing up all my volumes. 

Since I’m using Longhorn with k3s, it was straightforward to back everything up to S3. 

I also had all of my .yaml files backed up as well. After that, I simply formatted the NVMe drives on all four of my Raspberry Pis and set up HA k3s from scratch. 

Everything is now running cleanly with the fresh setup.
